### PR TITLE
[Algorithm] Support Rotated ATSS-OBB (CVPR'20)

### PR DESCRIPTION
**Pull request by yangxue0827**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Support [Rotated ATSS-OBB](https://arxiv.org/abs/1912.02424) (CVPR'20) for https://github.com/open-mmlab/mmrotate/issues/113.

## Modification
- Add new head: [rotated_atss_head](https://github.com/open-mmlab/mmrotate/blob/retinanet-atss/mmrotate/models/dense_heads/rotated_atss_head.py)
- Add new assigner: [atss_obb_assigner](https://github.com/open-mmlab/mmrotate/blob/retinanet-atss/mmrotate/core/bbox/assigners/atss_obb_assigner.py)
- Add new configs: [rotated_atss_hbb_r50_fpn_1x_dota_oc](https://github.com/open-mmlab/mmrotate/blob/retinanet-atss/configs/rotated_atss/rotated_atss_hbb_r50_fpn_1x_dota_oc.py), [rotated_atss_obb_r50_fpn_1x_dota_le90](https://github.com/open-mmlab/mmrotate/blob/retinanet-atss/configs/rotated_atss/rotated_atss_obb_r50_fpn_1x_dota_le90.py), [rotated_atss_obb_r50_fpn_1x_dota_le135](https://github.com/open-mmlab/mmrotate/blob/retinanet-atss/configs/rotated_atss/rotated_atss_obb_r50_fpn_1x_dota_le135.py)
- Update documentations: [model_zoo_en](https://github.com/open-mmlab/mmrotate/blob/main/docs/en/model_zoo.md), [model_zoo_zh-cn](https://github.com/yangxue0827/mmrotate/blob/main/docs/zh_cn/model_zoo.md), [README](https://github.com/open-mmlab/mmrotate/blob/main/README.md), [README_zh-CN](https://github.com/open-mmlab/mmrotate/blob/main/README_zh-CN.md)

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zytx121** [comment]
We should add `ATSSObbAssigner`, `ATSSKldAssigner` and `CSLCoder` in `mmrotate/mmrotate/core/bbox/__init__.py` so that the doc API can get the class.

File: configs/rotated_atss/metafile.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    In Collection: rotated_atss
+    Config: configs/rotated_atss/rotated_atss_hbb_r50_fpn_1x_dota_oc.py
+    Metadata:
+      Training Data: DOTAv1.0
+    Results:
+      - Task: Oriented Object Detection
+        Dataset: DOTAv1.0
+        Metrics:
+          mAP: 65.59
+    Weights: https://download.openmmlab.com/mmrotate/v0.1.0/rotated_atss/rotated_atss_hbb_r50_fpn_1x_dota_oc/rotated_atss_hbb_r50_fpn_1x_dota_oc-eaa94033.pth
```
Comment (zytx121) [inline configs/rotated_atss/metafile.yml:27]: v0.2.0

Comment (yangxue0827) [reply configs/rotated_atss/metafile.yml:27]: okay, i will give you the model and log, then fix it according to your final feedback.

**Comment by liuyanyi** [comment]
I noticed that most of the functions in `RotatedATSSHead` and `RotatedRetinaHead`  share the same code. Maybe it is possible to inherit directly from the `RotatedRetinaHead` and override the `get_targets` and `_get_targets_single` function. That will simplify the code.

**Review by liuyanyi — APPROVED** [review]
LGTM

**Review by zytx121 — APPROVED** [review]
LGTM

**Comment by ZwwWayne** [comment]
Is this PR ready for merge? I find it is marked as draft and I just mark it as ready by accident.

**Comment by yangxue0827** [comment]
This PR is ready to be merged. @ZwwWayne

File: configs/rotated_atss/metafile.yml (hunk 1 above)
Comment (zytx121) [reply configs/rotated_atss/metafile.yml:27]: Still v0.1.0

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**